components/tools/OmeroPy/src/omero/plugins/sessions.py

```diff
@@ -511,15 +511,13 @@ def login(self, args):
             if not pasw:
                 pasw = os.getenv("OMERO_PASSWORD")
 
-            if not pasw:
-                # Note: duplicating the `not pasw` check here
-                # for an overall nicer error message.
-                self._require_tty("cannot request password")
-
             tries = 3
             while True:
                 try:
                     if not pasw:
+                        # Handle absent and incorrect passwords in
+                        # non-interactive mode
+                        self._require_tty("cannot request password")
```

Comment:
maybe can still be outside the try/catch?

Reply:
It needs to be inside `if not pasw` to handle the case where a valid password is given on the command line

Reply:
Oh, doh, of course, sorry for the noise.